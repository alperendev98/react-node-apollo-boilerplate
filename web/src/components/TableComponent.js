import React from 'react';
import { getBEMClasses } from '../helpers/cssClassesHelper';

import '../styles/components/tableComponent.css';

import downloadIcon from '../assets/images/table-download.svg';
import uploadIcon from '../assets/images/table-upload.svg';
import historyIcon from '../assets/images/table-history.svg';

const BaseClasses = getBEMClasses(['base']);
const TableClasses = getBEMClasses(['table']);

class TableComponent extends React.Component {
	constructor(props) {
		super(props);
	}

	componentDidUpdate() {}

	render() {
		return (
			<div className={TableClasses('container')}>
				<div className={TableClasses('header')}>
					<div className={TableClasses('header-search')}>
						<input type="text" className={TableClasses('search-box')} placeholder="Search" />
					</div>
					<div className={TableClasses('header-buttons')}>
						<div className={TableClasses('header-button--download')}>
							<img src={downloadIcon} className={TableClasses('header-button--image')} />
						</div>
						<div className={TableClasses('header-button--upload')}>
							<img src={uploadIcon} className={TableClasses('header-button--image')} />
						</div>
						<div className={TableClasses('header-button--new')}>
							<img src={historyIcon} className={TableClasses('header-button--image')} />
						</div>
					</div>
				</div>
				<div className={TableClasses('content-header')}>
					<div className={TableClasses('table-header-column')}>
						<input type="checkbox" />
					</div>
					<div className={TableClasses('table-header-column')}>Name</div>
					<div className={TableClasses('table-header-column')}>Contact Info</div>
					<div className={TableClasses('table-header-column')}>Numberated Field</div>
					<div className={TableClasses('table-header-column')}>Other Field</div>
				</div>
				<div className={TableClasses('row')}>
					<div className={TableClasses('column')}>
						<input type="checkbox" />
					</div>
					<div className={TableClasses('column')}>Alliant Energy-Iowa</div>
					<div className={TableClasses('column')}>
						319-385-1577 Customer Service or ask for Sheryl Dorn, sdorn@accessenergycoop.com
					</div>
					<div className={TableClasses('column')}>9.4</div>
					<div className={TableClasses('column')}>Other Data</div>
				</div>
			</div>
		);
	}
}

export default TableComponent;
