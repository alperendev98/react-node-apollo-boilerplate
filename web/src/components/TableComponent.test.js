import React from 'react';
import { mount, shallow } from 'enzyme';
import TableComponent from './TableComponent';

describe('TableComponent', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallow(<TableComponent />);
	});

	it('shallowly renders TableComponent without crashing', () => {
		expect(wrapper.length).toEqual(1);
	});

	it('should match previos snapshot for TableComponent', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
