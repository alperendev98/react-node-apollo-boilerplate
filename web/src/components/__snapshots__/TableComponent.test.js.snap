// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableComponent should match previos snapshot for TableComponent 1`] = `
<div
  className="table__container"
>
  <div
    className="table__header"
  >
    <div
      className="table__header-search"
    >
      <input
        className="table__search-box"
        placeholder="Search"
        type="text"
      />
    </div>
    <div
      className="table__header-buttons"
    >
      <div
        className="table__header-button--download"
      >
        <img
          className="table__header-button--image"
          src="table-download.svg"
        />
      </div>
      <div
        className="table__header-button--upload"
      >
        <img
          className="table__header-button--image"
          src="table-upload.svg"
        />
      </div>
      <div
        className="table__header-button--new"
      >
        <img
          className="table__header-button--image"
          src="table-history.svg"
        />
      </div>
    </div>
  </div>
  <div
    className="table__content-header"
  >
    <div
      className="table__table-header-column"
    >
      <input
        type="checkbox"
      />
    </div>
    <div
      className="table__table-header-column"
    >
      Name
    </div>
    <div
      className="table__table-header-column"
    >
      Contact Info
    </div>
    <div
      className="table__table-header-column"
    >
      Numberated Field
    </div>
    <div
      className="table__table-header-column"
    >
      Other Field
    </div>
  </div>
  <div
    className="table__row"
  >
    <div
      className="table__column"
    >
      <input
        type="checkbox"
      />
    </div>
    <div
      className="table__column"
    >
      Alliant Energy-Iowa
    </div>
    <div
      className="table__column"
    >
      319-385-1577 Customer Service or ask for Sheryl Dorn, sdorn@accessenergycoop.com
    </div>
    <div
      className="table__column"
    >
      9.4
    </div>
    <div
      className="table__column"
    >
      Other Data
    </div>
  </div>
</div>
`;
